releasenotes/notes/add-passmanager-module-3ae30cff52cb83f1.yaml

```
@@ -3,24 +3,93 @@ features:
+    however, now running pass manager becomes a stateless controller object.
+  - |
+    Transpiler :class:`qiskit.transpiler.PassManager` is now a subclass of
+    the :class:`qiskit.passmanager.BasePassManager`.
+    There is no API break at public member level due to this class hierarchy change.
   - |
     New error baseclass :class:`~qiskit.passmanager.PassManagerError` is introduced.
     This will replace :class:`~qiskit.transpiler.TranspilerError` raised in the
-    pass handling machinery. The TranspilerError is now only used for the errors
-    related to the failure in handling the quantum circuit or DAG circuit object.
+    task handling machinery. The TranspilerError is now only used for the errors
+    related to the failure in handling of the quantum circuit or DAG circuit object.
     Note that the TranspilerError can be caught by the PassManagerError
     because of their class hierarchy. For backward compatibility,
     :class:`qiskit.transpiler.PassManager` catches PassManagerError and
     re-raises the TranspilerError. This error replacement will be dropped in future.
+  - |
+    Use of :class:`.FencedObject` in the pass manager framework is removed.
+    These wrapper class cannot protect mutable object attribute from modification,
+    and protection doesn't matter as long as the code is properly implemented;
+    analysis passes should not modify an input IR,
+    controllers should not update the property set, and so forth.
+    Implementation of the proper code is the responsibility of pass manager developer.
+deprecations:
+  - |
+    The flow controller factory method :meth:`.FlowController.controller_factory` is deprecated,
+    as we are also going to deprecate task construction with
+    keyword arguments in the :meth:`.BasePassManager.append` method.
+    Controllers must be explicitly instantiated and appended to the pass manager.
+    For example, conventional syntax
+
+    .. code-block:: python
+
+        pm.append([task1, task2], condition=lambda x: x["value1"] > 10)
+
+    must be replaced with
+
+    .. code-block:: python
+
+        controller = ConditionalController([task1, task2], condition=lambda x: x["value1"] > 10)
+        pm.append(controller)
+
+    The latter must allow more precise control on the order of controllers
+    especially when multiple keyword arguments are specified together.
+  - |
+    The :meth:`.FlowControllerLinear.append`, :meth:`.DoWhileController.append`,
+    and :meth:`.ConditionalController.append` method are all deprecated immediately.
+    The construction of pass manager task pipeline is the role of :class:`.BasePassManager`,
+    and individual flow controller doesn't need to implement method like this.
+  - |
+    The all attribute and variable name :code:`passes` are replaced with :code:`tasks`
+    in the :mode:`passmanager` module. Note that a task must indicate a union of pass and controller,
```

This should fix the docs build error in the most recent CI run:

```suggestion
    in the :mod:`qiskit.passmanager` module. Note that a task must indicate a union of pass and controller,
```

---

thanks!